### PR TITLE
common/thanos: sort query store apis

### PR DESCRIPTION
**Pull request by richardtief**
Query store api's are added to the service discovery file in different orders. This quickly becomes disorganized if many targets are involved (global). This PR sorts the targets alphabetically.

### CODE REVIEW

File: common/thanos/templates/_thanos_helpers.tpl [hunk 1]
```diff
@@ -130,28 +130,47 @@ cluster.local
...
 {{- $store := $service.metadata.name }}
-{{ printf "- dnssrvnoa+_grpc._tcp.%s.%s.svc.%s" $store $service.metadata.namespace (include "clusterDomainOrDefault" $root) | indent 2 }}
+{{- $storeItem := printf "dnssrvnoa+_grpc._tcp.%s.%s.svc.%s" $store $service.metadata.namespace (include "clusterDomainOrDefault" $root) -}}
+{{- $clusterList = append $clusterList $storeItem -}}
 {{- end }}
 {{- end }}
+{{- $sortedClusterList := $clusterList | sortAlpha -}}
+{{- range $queryStoreAPI := $sortedClusterList }}
+  - {{ $queryStoreAPI }}
+{{- end }}
```
Comment (viennaa) [inline common/thanos/templates/_thanos_helpers.tpl:144]: Would simplify this:

```suggestion
{{- range $clusterList | sortAlpha -}}
  - {{ . }}
{{- end }}
```

File: common/thanos/templates/_thanos_helpers.tpl [hunk 2]
```diff
@@ -130,28 +130,47 @@ cluster.local
...
 - targets:
+{{- $regionalList := list }}
 {{- range $cluster := $root.Values.queryStoreAPIs }}
-  - thanos-{{ $cluster }}-grpc.{{ $root.Values.global.region }}.{{ $root.Values.global.tld }}:443
+{{- $storeItem := printf "thanos-%s-grpc.%s.%s" $cluster $root.Values.global.region $root.Values.global.tld -}}
+{{- $regionalList = append $regionalList $storeItem -}}
+{{- end }}
+{{- $sortedRegionalList := $regionalList | sortAlpha -}}
+{{- range $queryStoreAPI := $sortedRegionalList }}
+  - {{ $queryStoreAPI }}:443
```
Comment (viennaa) [inline common/thanos/templates/_thanos_helpers.tpl:173]:
```suggestion
{{- range $regionalList | sortAlpha -}}
  - {{ . }}:443
```

**Review by viennaa — CHANGES_REQUESTED** [review]
Like the idea, makes it easier to read.

Would suggest to directly use the sorted list in a range query.

File: common/thanos/templates/_thanos_helpers.tpl [hunk 3]
```diff
@@ -130,28 +130,45 @@ cluster.local
...
+{{- $globalList = append $globalList $storeItem -}}
+{{- end -}}
+{{- end -}}
+{{- range $store := $root.Values.query.stores -}}
+{{- $storeItem := printf "%s.%s" $store $root.Values.global.tld -}}
+{{- $globalList = append $globalList $storeItem -}}
+{{- end -}}
+{{- $sortedGlobalList := $globalList | sortAlpha -}}
+{{- range $queryStoreAPI := $sortedGlobalList }}
+  - {{ $queryStoreAPI }}:443
```
Comment (richardtief) [inline common/thanos/templates/_thanos_helpers.tpl:160]:
```suggestion
{{- range $globalList | sortAlpha -}}
  - {{ . }}
{{- end }}
```

File: common/thanos/templates/_thanos_helpers.tpl [hunk 4]
```diff
@@ -130,28 +130,45 @@ cluster.local
...
+{{- $storeItem := printf "thanos-%s-grpc.%s.%s" $cluster $region $root.Values.global.tld -}}
+{{- $globalList = append $globalList $storeItem -}}
+{{- end -}}
+{{- end -}}
+{{- range $store := $root.Values.query.stores -}}
+{{- $storeItem := printf "%s.%s" $store $root.Values.global.tld -}}
+{{- $globalList = append $globalList $storeItem -}}
+{{- end -}}
+{{- range $globalList | sortAlpha -}}
+  - {{ . }}
```
Comment (richardtief) [inline common/thanos/templates/_thanos_helpers.tpl:159]:
```suggestion
  - {{ . }}:443
```

**Pull request merged**